### ISSUE TITLE
GHA: Collect test statistics

### ISSUE THREAD

**Issue by lenaschoenburg** [camunda/camunda#9132]
Jenkins collects statistical information about our tests:

* Time to run each test
* Total number of tests
* Number of flaky tests
* Total job runtime

The flaky test statistics are collected in BigTable while the total number of tests and the timings are stored by Jenkins directly. Maybe we can find a solution where we just analyze the test reports once and extract all necessary information from there. It's not yet clear to me where we would display or store the results 

@npepinpe brought up that the total number of tests is something that we should make very visible since we don't want to accidentally skip running tests (as has happened in the past).

blocked by https://jira.camunda.com/browse/INFRA-3363

**Comment by npepinpe** [comment]
Some options:

- https://github.com/marketplace/testspace-com/plan
- https://github.com/marketplace/meercode-ci-monitoring
- https://github.com/marketplace/thundra-foresight

All of which should be free for open source projects, I think.

**Comment by npepinpe** [comment]
Seems like DataDog also offers this, but DataDog in my experience is :money_mouth_face: (though it's also very, very, very good): https://www.datadoghq.com/blog/datadog-ci-visibility/

**Comment by oleschoenburg** [comment]
That does sound like exactly what we'd need. Apparently it starts at 20$ per committer per month: https://www.datadoghq.com/pricing/?product=ci-visibility#ci-visibility

**Comment by npepinpe** [comment]
Yeah, for an open source project this may be less than ideal :smile: DataDog is very good, just always very pricey

**Comment by npepinpe** [comment]
In the end, we can keep doing what we're doing, and simply send the statistics to BigQuery - but instead of only flaky tests, we send more general statistics. Grafana 8.x has official support for BigQuery, so we could import data in there for our dashboards without having to learn a new tool to create dashboards. I guess that would be one option if none of the others work for us.

**Comment by npepinpe** [comment]
Or let's keep it simple, and let's just send our statistics as we do now. Maybe infra can help us adapt how Jenkins connects to BigQuery to GHA. We can add some basic statistics later (# of tests) once that works.

**Comment by npepinpe** [comment]
Also, supposedly, SonarCloud should be able to at least count the number of tests. So we have our basic stats imo

**Comment by npepinpe** [comment]
I think SonarCloud could do this. See https://docs.sonarcloud.io/enriching/test-coverage/test-execution-parameters/ - I don't think we need to set a target, but just reporting it might be good enough for now.

**Comment by Zelldon** [comment]
Out of interest what is the usage benefit of this? Isn't something like this https://github.com/camunda/zeebe/pull/9450#issuecomment-1137447651 enough for now?

**Comment by npepinpe** [comment]
What's missing now is the build status/flaky test data collection which feeds our flaky test dashboard. I make use of it to track where to focus efforts, so I would like to have this at least.

**Comment by menski** [comment]
Moved to blocked as we need support by infra

**Comment by menski** [comment]
Infra ticket: https://jira.camunda.com/browse/INFRA-3363

**Comment by cmur2** [comment]
FYI: Your input is needd on [INFRA-3363](https://jira.camunda.com/browse/INFRA-3363?focusedCommentId=247258&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-247258), thanks!

**Comment by npepinpe** [comment]
See:

- https://github.com/camunda/infra-global-github-actions/tree/main/submit-build-status
- https://confluence.camunda.com/pages/viewpage.action?spaceKey=HAN&title=CI+Analytics#CIAnalytics-Usecase:BuildstatusstatisticsonGithubActions
- https://github.com/camunda/zeebe/pull/10205
